### PR TITLE
feat(schematics): add `tsconfig` for `ng-packagr`

### PR DESCRIPTION
**Pull request by alan-agius4**
Add a tsconfig build for ng-packagr as some users are having a hard time to create a valid one + I dont like the fact of a tsconfig simply for lint.

Relates: https://github.com/dherges/ng-packagr/issues/797

### CODE REVIEW

File: packages/schematics/angular/library/index.ts [hunk 1]
```diff
@@ -156,7 +157,7 @@ function addAppToWorkspaceFile(options: LibraryOptions, workspace: WorkspaceSche
         builder: '@angular-devkit/build-angular:tslint',
         options: {
           tsConfig: [
-            `${projectRoot}/tsconfig.lint.json`,
+            `${projectRoot}/tsconfig.build.json`,
```
Comment (filipesilva) [inline packages/schematics/angular/library/index.ts]: In our tests we use a `tsconfig.lib.json`, as a counterpart to `tsconfig.app.json`. Can you change this filename please?

**Comment by filipesilva** [comment]
CI is failing at least on windows with what seems to be a legit error:

```
1) NgPackagr Builder works
  - Failed: C:/projects/devkit/node_modules/rxjs/internal/Subscription.d.ts:14:1: Error encountered in metadata generated for exported symbol 'Subscription': 
   C:/projects/devkit/node_modules/rxjs/internal/Subscription.d.ts:16:12: Metadata collected contains an error that will be reported at runtime: Only initialized variables and constants can be referenced because the value of this variable is needed by the template compiler.
    {"__symbolic":"error","message":"Variable not initialized","line":15,"character":11}
  Error: C:/projects/devkit/node_modules/rxjs/internal/Subscription.d.ts:14:1: Error encountered in metadata generated for exported symbol 'Subscription': 
```

Does this look familiar?

**Comment by alan-agius4** [comment]
Not really familiar, but I'll debug it, as I managed to replicate it on mac as well.

**Comment by alan-agius4** [comment]
Not really familiar, but I'll debug it, as I managed to replicate it on mac as well.

But i think it's related to the `fullTemplateTypeCheck` which previously was not enabled.

**Comment by alan-agius4** [comment]
@filipesilva this look like it's an angular compiler issue https://github.com/angular/angular/issues/18867

File: packages/schematics/angular/library/files/__projectRoot__/tsconfig.lib.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "extends": "<%= projectRoot.split('/').map(x => '..').join('/') %>/tsconfig.json",
+  "compilerOptions": {
+    "outDir": "",
```
Comment (filipesilva) [inline packages/schematics/angular/library/files/__projectRoot__/tsconfig.lib.json]: We should add an outdir for when ngc is used directly (we were debugging the error message as well and ran into it).

On another node, we could not reproduce the error locally... were you able to?

Comment (alan-agius4) [reply packages/schematics/angular/library/files/__projectRoot__/tsconfig.lib.json]: Good point, re the `outDir`,  actually I got mislead by the `hello world app` failing unit test. I am still waiting for the tests to run. Is there any short cut to run a specific set of N tests?

Comment (alan-agius4) [reply packages/schematics/angular/library/files/__projectRoot__/tsconfig.lib.json]: Update I did manage to replicate it 
screen shot 2018-04-25 at 20 59 33

I will have to look at this tomorrow.

**Comment by alan-agius4** [comment]
@filipesilva Build Packager CI is now green 😎, bumping the packages solved the issue

But now I am having, which doesn't seem to be related.
```
Browser Builder works with service worker
  - Expected object not to have properties
      navigationUrls: [ Object({ positive: true, regex: '^\/.*$' }), Object({ positive: false, regex: '^\/(?:.+\/)?[^\/]*\.[^\/]*$' }), Object({ positive: false, regex: '^\/(?:.+\/)?[^\/]*__[^\/]*$' }), Object({ positive: false, regex: '^\/(?:.+\/)?[^\/]*__[^\/]*\/.*$' }) ]
  Error: Expected object not to have properties
```

**Comment by filipesilva** [comment]
Great work in this PR @alan-agius4, thanks for also fixing the updated SW test! 

There's still an error but it seems like a escaping error:
```
1) Browser Builder works with service worker
  - Expected $.navigationUrls[0].regex = '^\/.*$' to equal '^/.*$'.
  Expected $.navigationUrls[1].regex = '^\/(?:.+\/)?[^\/]*\.[^\/]*$' to equal '^/(?:.+/)?[^/]*.[^/]*$'.
  Expected $.navigationUrls[2].regex = '^\/(?:.+\/)?[^\/]*__[^\/]*$' to equal '^/(?:.+/)?[^/]*__[^/]*$'.
  Expected $.navigationUrls[3].regex = '^\/(?:.+\/)?[^\/]*__[^\/]*\/.*$' to equal '^/(?:.+/)?[^/]*__[^/]*/.*$'.
  Error: Expected $.navigationUrls[0].regex = '^\/.*$' to equal '^/.*$'.

2) Browser Builder works with service worker and baseHref
  - Expected $.navigationUrls[0].regex = '^\/.*$' to equal '^/.*$'.
  Expected $.navigationUrls[1].regex = '^\/(?:.+\/)?[^\/]*\.[^\/]*$' to equal '^/(?:.+/)?[^/]*.[^/]*$'.
  Expected $.navigationUrls[2].regex = '^\/(?:.+\/)?[^\/]*__[^\/]*$' to equal '^/(?:.+/)?[^/]*__[^/]*$'.
  Expected $.navigationUrls[3].regex = '^\/(?:.+\/)?[^\/]*__[^\/]*\/.*$' to equal '^/(?:.+/)?[^/]*__[^/]*/.*$'.
  Error: Expected $.navigationUrls[0].regex = '^\/.*$' to equal '^/.*$'.
```

**Comment by alan-agius4** [comment]
@filipesilva, I pushed a quick tentative fix for the escape chars, hopefully it fixes it. I am off to sleep, I'll re-check tmr :)

**Comment by filipesilva** [comment]
Oh and before I forget, if you only want to run one of those test just add a `fit` instead of `it` and run `npm run test-large`.

**Comment by filipesilva** [comment]
It passed everything, great job on this PR!

**Pull request merged**